### PR TITLE
Add Show instances for a couple of basic types

### PR DESCRIPTION
**Pull request by Yuuriosity**
Instances for basic types plus a couple of little string manipulation helpers for future instances.

### CODE REVIEW

File: src/Data/Char/Instances/Show.agda [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{-# OPTIONS --safe #-}
+module Data.Char.Instances.Show where
+
+open import Foundations.Base
+
+open import Meta.Show
+
+open import Agda.Builtin.String
```
Comment (cmcmA20) [inline src/Data/Char/Instances/Show.agda:8]: open import `Data.String.Base`

File: src/Data/Float/Instances/Show.agda [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{-# OPTIONS --safe #-}
+module Data.Float.Instances.Show where
+
+open import Meta.Show
+
+open import Agda.Builtin.Float
```
Comment (cmcmA20) [inline src/Data/Float/Instances/Show.agda:6]: Same, should import the corresponding parent module `Data.Float.Base`

File: src/Data/String/Base.agda
Comment (cmcmA20) [inline src/Data/String/Base.agda:1]: Just rename `primStringAppend` to `++ₛ`

File: src/Data/Sum/Instances/Everything.agda
Comment (cmcmA20) [inline src/Data/Sum/Instances/Everything.agda:1]: `Everything.agda`  files are generated by build script and .gitignored

File: src/Foundations/Sigma.agda
Comment (cmcmA20) [inline src/Foundations/Sigma.agda:1]: This should go directly into `Meta.Show`.

Instances for basic typeformers like pi and sigma live at the definition site of the corresponding typeclass/tactic. Also importing stuff outside of `Foundations` is explicitly forbidden.

File: src/Data/Fin/Instances/Show.agda [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{-# OPTIONS --safe #-}
+module Data.Fin.Instances.Show where
+
+open import Meta.Show
+
+open import Agda.Builtin.String
```
Comment (cmcmA20) [inline src/Data/Fin/Instances/Show.agda:6]: same

File: src/Data/Int/Instances/Show.agda [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{-# OPTIONS --safe #-}
+module Data.Int.Instances.Show where
+
+open import Meta.Show
+
+open import Agda.Builtin.Int
```
Comment (cmcmA20) [inline src/Data/Int/Instances/Show.agda:6]: same

File: src/Data/Nat/Instances/Show.agda [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{-# OPTIONS --safe #-}
+module Data.Nat.Instances.Show where
+
+open import Meta.Show
+
+open import Agda.Builtin.String
```
Comment (cmcmA20) [inline src/Data/Nat/Instances/Show.agda:6]: same

File: src/Data/String/Base.agda [hunk 6]
```diff
@@ -7,6 +7,9 @@ open import Agda.Builtin.String public
     ( primStringUncons   to uncons
     ; primStringToList   to string→list
     ; primStringFromList to list→string
-    ; primStringAppend   to concat-str
```
Comment (cmcmA20) [inline src/Data/String/Base.agda:10]: Just rename `primStringAppend` to `_++ₛ_`

Comment (Yuuriosity) [reply src/Data/String/Base.agda:10]: > Just rename `primStringAppend` to `_++ₛ_`

That was my first try, but it then warns for set fixity for an operator defined in another module.

Comment (cmcmA20) [reply src/Data/String/Base.agda:10]: Ah okay then!

Comment (cmcmA20) [reply src/Data/String/Base.agda:10]: I was mistaken, see the [manual](https://agda.readthedocs.io/en/v2.6.3.20230805/language/mixfix-operators.html)
>Fixities can be changed when importing with a renaming directive:
```
open M using (_∙_)
open M renaming (_∙_ to infixl 10 _*_)
```

**Comment by cmcmA20** [comment]
Merged manually, thank your for contribution.

**Pull request closed without merging**